dashboard/common/db_util.py

```diff
@@ -12,16 +13,29 @@
 
 logger = logging.getLogger(__name__)
 
+def canvas_id_to_prefixed_id(canvas_id):
+    # Add the DATA_WAREHOUSE_ID_PREFIX and to the id
+    canvas_id = str(canvas_id)
+    prefixed_id = str(settings.DATA_WAREHOUSE_ID_PREFIX)[0:-len(canvas_id)] + canvas_id
```

Comment:
This particular code end up creating a wrong course id. This particular function is called from from admin view for adding courses from django admin view and that statement will end up in wrong long canvas id.  this `[0:-len(canvas_id)]` needs to removed in order to have right id format. what was the reason behind substring in the above statement?
```
canvas_id = '235999'
DATA_WAREHOUSE_ID_PREFIX=17700000000
prefixed_id = str(DATA_WAREHOUSE_ID_PREFIX)[0:-len(canvas_id)] + canvas_id
print(prefixed_id) 

o/p: 17700235999 instead of 17700000000235999
```

Reply:
> Note: this is a breaking change to DATA_WAREHOUSE_ID_PREFIX. Existing installation will need to update the value with the correct number of additional 0s added to the end

Try changing your `DATA_WAREHOUSE_ID_PREFIX` from `17700000000` to `17700000000000000` (six extra zeros).

I switched to using a safer approach to adding the prefix instead of simply putting a static prefix in front of the canvas id since doing that won't work for canvas id of different orders (ex: canvas id 100 and 1000 won't be prefixed properly if you simply put the `17700000000` in front of both of them). instead it will insert the canvas id into the prefix string as you saw (ex: when using `17700000000000000` it will do `17700000000000100` or `17700000000001000` for ids 100 and 1000)

Reply:
I'll also rename the prefix variable to template since it isn't really a straight prefix anymore

Reply:
@andrew-gardener yes, the name(prefix) is what caused the confusion. Changing it as `DATA_WAREHOUSE_ID_TEMPLATE` or `DATA_WAREHOUSE_EMPTY_ID` will make sense.

Reply:
Renamed to `DATA_WAREHOUSE_ID_TEMPLATE` and improved the comments explaining how `canvas_id_to_prefixed_id` and `prefixed_id_to_canvas_id` work